File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/javaNapakalaki/NapakalakiGame/BadConsequence.java
package NapakalakiGame;
import java.lang.*;

import java.util.ArrayList;

/**
 *
 * @author manjaro
 */

public abstract class BadConsequence {

    private String text="";
    private int levels;
    private static final int MAXTREASURES = 10;

    public BadConsequence(String text, int levels){
        this.text = text;
        this.levels = levels;
    }

    public String getText() {
        return text;
    }


    public int getLevels() {
        return levels;
    }

    public void setLevels(int l){
         this.levels = l;
    }

    public boolean isEmpty(){
         boolean empty = false;
         if(text == "" && levels == 0 ){
            empty = true;
         }
      return empty;
     }

    public abstract boolean isDeath();

    public abstract void setnVisibleTreasures(int n);

    public abstract void setnHiddenTreasures(int n);

    public abstract int getnVisibleTreasures();

    public abstract int getnHiddenTreasures();

    public abstract void substractVisibleTreasure(Treasure t);

    public abstract void substractHiddenTreasure(Treasure t);

    public abstract  BadConsequence adjustToFitTreasureLists(ArrayList<Treasure> v , ArrayList<Treasure> h);

    public abstract String toStringPendingVisible();

    public abstract String toStringPendingOculto();


}

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/rubyNapakalaki/bad_consequence.rb
#encoding: utf-8

class Bad_consequence
  attr_accessor :text ,:levels



  def initialize (t, l)
    @text = t
    @levels = l
  end


  def to_s()
    "Texto: #{@text} Niveles perdidos: #{@levels} "
  end

  def setLevels(l)

    @levels = l

  end

  def is_empty

    empty = false
         if @text == "" and @levels == 0
              empty = true
         end
    empty

  end

  def getLevels
    @levels
  end

  def getText
    @text
  end

  def substract_visible_treasure(t)
    raise NotImplementedError.new
  end

  def substract_hidden_treasure(t)
    raise NotImplementedError.new
  end

  def is_dead
    raise NotImplementedError.new
  end

  def setnVisibleTreasures(n)
      raise NotImplementedError.new
  end

  def setnHiddenTreasures(n)
      raise NotImplementedError.new
  end

  def getnVisibleTreasures
      raise NotImplementedError.new
  end

  def getnHiddenTreasures
      raise NotImplementedError.new
  end




  def adjust_to_fit_treasure_list(v, h)
      raise NotImplementedError.new

  end

end
